smartsim/_core/entrypoints/telemetrymonitor.py

```diff
@@ -0,0 +1,687 @@
+
+    :param observer: (optional) the watchdog Observer instance
+    :type observer: t.Optional[BaseObserver]
+    :param action_handler: The manifest event processor instance
+    :type action_handler: ManifestEventHandler"""
+    if can_shutdown(action_handler):
+        observer.stop()  # type: ignore[no-untyped-call]
+
+
+def event_loop(
+    observer: BaseObserver,
+    action_handler: ManifestEventHandler,
+    frequency: t.Union[int, float],
+    logger: logging.Logger,
+) -> None:
+    """Executes all attached timestep handlers every <frequency> seconds
+
+    :param observer: (optional) a preconfigured watchdog Observer to inject
+    :type observer: t.Optional[BaseObserver]
+    :param action_handler: The manifest event processor instance
+    :type action_handler: ManifestEventHandler
+    :param frequency: frequency (in seconds) of update loop
+    :type frequency: t.Union[int, float]
+    :param experiment_dir: the experiement directory to monitor for changes
+    :type experiment_dir: pathlib.Path
+    :param logger: a preconfigured Logger instance
+    :type logger: logging.Logger"""
+    while observer.is_alive():
+        timestamp = get_ts()
+        logger.debug(f"Telemetry timestep: {timestamp}")
+        action_handler.on_timestep(timestamp)
+        time.sleep(frequency)
+
+        shutdown_when_completed(observer, action_handler)
+
+
+def main(
+    frequency: t.Union[int, float],
+    experiment_dir: pathlib.Path,
+    logger: logging.Logger,
+    observer: t.Optional[BaseObserver] = None,
+) -> int:
+    """Setup the monitoring entities and start the timer-based loop that
+    will poll for telemetry data
+
+    :param frequency: frequency (in seconds) of update loop
+    :type frequency: t.Union[int, float]
+    :param experiment_dir: the experiement directory to monitor for changes
+    :type experiment_dir: pathlib.Path
+    :param observer: (optional) a preconfigured Observer to inject
+    :type observer: t.Optional[BaseObserver]
+    """
+    manifest_relpath = pathlib.Path(TELMON_SUBDIR) / MANIFEST_FILENAME
+    manifest_path = experiment_dir / manifest_relpath
+    monitor_pattern = str(manifest_relpath)
+
+    logger.info(
+        f"Executing telemetry monitor with frequency: {frequency}s"
+        f", on target directory: {experiment_dir}"
+        f" matching pattern: {monitor_pattern}"
+    )
+
+    log_handler = LoggingEventHandler(logger)  # type: ignore
+    action_handler = ManifestEventHandler(monitor_pattern, logger)
+
+    if observer is None:
+        observer = Observer()
+
+    try:
+        if manifest_path.exists():
+            # a manifest may not exist depending on startup timing
+            action_handler.process_manifest(str(manifest_path))
+
+        observer.schedule(log_handler, experiment_dir, recursive=True)  # type:ignore
+        observer.schedule(action_handler, experiment_dir, recursive=True)  # type:ignore
```

Comment:
I'm guessing these suppressions are all coming from "untyped fn calls in typed context" errors. If you want feel free to just disable at a global level!

Reply:
I'm divided.... I think I'd rather leave it on globally and catch missing stubs.